src/lib/core/data-table/row.ts

```diff
@@ -89,6 +89,29 @@ export class CdkRowDef extends BaseRowDef {
   }
 }
 
+/** Context provided to the row cells */
+export class CdkCellOutletRowContext<T> {
+  /** Index location of the row that this cell is located within. */
+  index: number;
+
+  /** Length of the number of total rows. */
+  count: number;
+
+  constructor(public $implicit: T) {}
+
+  /** True if this cell is contained in the first row. */
+  get first(): boolean { return this.index === 0; }
+
+  /** True if this cell is contained in the last row. */
+  get last(): boolean { return this.index === this.count - 1; }
+
+  /** True if this cell is contained in a row with an even-numbered index. */
+  get even(): boolean { return this.index % 2 === 0; }
+
+  /** True if this cell is contained in a row with an odd-numbered index. */
+  get odd(): boolean { return !this.even; }
```

Comment:
These should be set statically; getters and setters generate a lot of code, so it's better to avoid them when they're not necessary